src/main/scala/org/deepdive/ddlog/DeepDiveLogPrettyPrinter.scala

```diff
@@ -78,51 +80,64 @@ object DeepDiveLogPrettyPrinter extends DeepDiveLogHandler {
         else value
       }
       case FuncExpr(function, args, agg) => {
-        val resolvedArgs = args map (x => printExpr(x))
+        val resolvedArgs = args map (x => print(x))
         s"${function}(${resolvedArgs.mkString(", ")})"
       }
-      case BinaryOpExpr(lhs, op, rhs) => s"(${printExpr(lhs)} ${op} ${printExpr(rhs)})"
-      case TypecastExpr(lhs, rhs) => s"(${printExpr(lhs)} :: ${rhs})"
+      case BinaryOpExpr(lhs, op, rhs) => s"(${print(lhs)} ${op} ${print(rhs)})"
+      case TypecastExpr(lhs, rhs) => s"(${print(lhs)} :: ${rhs})"
     }
   }
 
-  // print a condition
-  def printCond(cond: Cond) : String = {
-    cond match {
-      case ComparisonCond(lhs, op, rhs) => s"${printExpr(lhs)} ${op} ${printExpr(rhs)}"
-      case NegationCond(c) => s"[!${printCond(c)}]"
-      case CompoundCond(lhs, op, rhs) => {
-        op match {
-          case LogicOperator.AND => s"[${printCond(lhs)}, ${printCond(rhs)}]" 
-          case LogicOperator.OR  => s"[${printCond(lhs)}; ${printCond(rhs)}]"
+  // print a body
+  def print(b: Body) : String = {
+    def printAtom(a: Atom) = {
```

Comment:
Let's expose them.  Ideally, the `print(b: Body)` method with boring dispatch should be taken care by scala and be optional, but that's a shortcoming of JVM's ad-hoc polymorphism.